src/Controller/HomeController.php

```diff
@@ -22,12 +21,12 @@ public function home()
         ]);
     }
 
-    public function getArgs(): array
+    private function getArgs(): array
     {
-        $config =  App::get('config')['Api'];
-        $provider = $_GET['provider'] ?? $config['provider'];
-        $language = $_GET['language'] ?? $config['language'];
-        $since = $_GET['since'] ?? $config['since'];
+        $api =  $this->config['Api'];
```

Comment:
More direct hint: actually,  information from `$this->config['Api']` is **everything** you need from the `App` in the controllers, isn't it?

Reply:
I also need `$this->config['offset']`

Reply:
True, overlooked. Then, change my phrase to `$this->config` is **everything** you... 😉

Reply:
I won :fireworks: :fire: :laughing:

Reply:
You won the battle, but not the war ☝️ 🤣